### PR TITLE
Deprecate fixes for 2.11.x

### PR DESCRIPTION
**Pull request by tvierling**

### CODE REVIEW

**Comment by adriaanm** [comment]
Good catch! Review by @Ichoran to make sure it was not intentional.

File: src/library/scala/collection/mutable/AVLTree.scala [hunk 1]
```diff
@@ -14,8 +14,8 @@ package mutable
  * An immutable AVL Tree implementation formerly used by mutable.TreeSet
  *
  * @author Lucien Pereira
- * @deprecated("AVLTree and its related classes are being removed from the standard library since they're not different enough from RedBlackTree to justify keeping them.", "2.11.0")
  */
+@deprecated("AVLTree and its related classes are being removed from the standard library since they're not different enough from RedBlackTree to justify keeping them.", "2.11.0")
```
Comment (Ichoran) [inline src/library/scala/collection/mutable/AVLTree.scala]: This actually needs to be deprecated as of 2.11.2, since that's the earliest it will go in.

**Comment by Ichoran** [comment]
I am not sure whether Lock should be deprecated.  It is just the lightest-weight wrapper around the built-in synchronized locking.  The java.util locks referred to in the deprecation method are vastly heavier-weight.  I'll do some more digging and see if Lock is actually used by anyone anywhere.

**Comment by retronym** [comment]
IIRC I deprecated (or, tried to) it on the grounds of poor design (anyone can set `available`), absent documentation / tests, and perceived lack of use.

**Comment by tvierling** [comment]
As-is, it's pretty bad as far as design goes. Ideally such a Lock object would have a thunk-based method equivalent to the built-in `synchronized {}` construct, not explicit `acquire()`/`release()` calls, as the latter is designed to leave dangling locks.

Then, since we're talking about built-in `synchronized {}`, there's no need for an external class; just do `someObject.synchronized {}`.

If anything is currently using `scala.concurrent.Lock`, it is fostering broken semantics and should be fixed.

**Comment by tvierling** [comment]
(By contrast, an enrichment of `java.util.concurrent.locks.Lock` which exposes a thunk-passing method would be quite useful.)

**Comment by Ichoran** [comment]
Er, the whole point of having this wrapper is to convert synchronized { },
which you can just use plain, into a context where you need (and are
willing to worry about) dangling locks.

It's not something you'd want to use for safety, but for capability.

But not encouraging this sort of low-level dangerous fiddling is probably a
good idea, so we should probably deprecate regardless of usage.  (Just mark
it 2.11.2.)

**Comment by tvierling** [comment]
Well, sort of. It's simple enough that someone needing those specific semantics can rewrite it trivially.

I guess that my interpretation of "the Scala way" to approach these problems has become purely-functional in scope: the lock should be handled behind the scenes while executing a code block, which is precisely what `synchronized {}` does syntactically, anyway.

Will re-push the branch with markers retargeted to 2.11.2 and will drop the other pull request for 2.12.x since it'll get auto-synced later.

**Comment by Ichoran** [comment]
LGTM.

**Pull request merged**